enhancements/cluster-logging/cluster-logging-move-init-to-es-operator.md

```diff
@@ -0,0 +1,65 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+reviewers:
+  - "@bparees"
+  - "@jeremyeder"
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+approvers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@richm"
+creation-date: 2019-09-17
+last-updated: 2019-10-25
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-es-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of moving Elasticsearch initialization logic to the elasticsearch-operator is to reduce the number of times these init scripts executed.
+At the moment we keep these scripts in the ES image, and execute them everytime when ES pod starts. 
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that allows users to execute init procedures for the ES cluster
+* Allow users to choose what command has to be executed during the init process for FULL and ROLLING elasticsearch cluster restart
+
+## Proposal
+
+The following elasticsearch-operator changes will allow to execute init process for ES cluster only when we restarting the whole cluster, not it's pods:
+
+
+* Adding additional section to the elasticsearch-operator called `bootstrap`, which may look like this:
+
+    ```json
+    "bootstrap": {
```

Comment:
Let's be explicit about what happens when this segment is not defined.  I presume we will do nothing

Reply:
We can also execute the `init.sh` script from the image (hard-coded) as we do now if this section is empty. But I'd say if the `image` is provided, actions also have to be specified.

Reply:
i would say do thing is better by default. We will make changes to the EO to do these things